### PR TITLE
Add Contributing doc and release template

### PR DESCRIPTION
**Pull request by mikeyarce**
Addresses https://github.com/Parsely/wp-parsely/issues/212

## Description

- Adds Contributing.md which aims to help clearly lay out contribution guidelines and has examples for:
  - Reporting Bugs
  - Setting up environment
  - Installing dependencies
  - Contributing new features and patches
  - Coding Standards
  - Linting
  - Unit Tests
  - Building assets

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-template.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+about: Internally used for new releases
+title: Release x.y.z
+labels: 'Type: Maintenance'
+assignees: jblz, GaryJones, mikeyarce, hbbtstar
+
+---
+
+:warning: DO NOT MERGE (YET) :warning:
+
+[Remaining work for this Milestone]``(https://github.com/Parsely/wp-parsely/milestone/14)
```
Comment (jblz) [inline .github/ISSUE_TEMPLATE/release-template.md]: Did those backticks sneak in or do they serve a purpose?

```suggestion
[Remaining work for this Milestone](https://github.com/Parsely/wp-parsely/milestone/MILESTONE_ID_HERE)
```

File: .github/ISSUE_TEMPLATE/release-template.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+name: Release template
+about: Internally used for new releases
+title: Release x.y.z
+labels: 'Type: Maintenance'
+assignees: jblz, GaryJones, mikeyarce, hbbtstar
```
Comment (jblz) [inline .github/ISSUE_TEMPLATE/release-template.md]:
```suggestion
```

Bummer that assignees can be a group.  I'd probably leave it out so we don't have to change this to add / rm people

**Review by jblz — COMMENTED** [review]
Super awesome!

Left a couple of suggestions.

Personally, I think we can rm https://github.com/Parsely/wp-parsely/blob/add/plugin-processes/DEVELOP.md as well.

File: .github/ISSUE_TEMPLATE/release-template.md (hunk 1 above)
Comment (mikeyarce) [reply .github/ISSUE_TEMPLATE/release-template.md]: I blame my cat.

**Comment by mikeyarce** [comment]
Thanks @jblz - implemented suggestions! 😄

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+When creating a new issue, please add specific steps to reproduce the problem, upload any relevant screenshots, and describe what happened and what you expected would happen instead.
+
+## Setting up your environment
+
+### Minimum required versions
+
+* #### Node.js - LTS
+	Node.js is used in the build process of the Parse.ly plugin. If it's not already installed on your system, you can [visit the Node.js website and install the latest Long Term Support (LTS) version.](https://nodejs.org/)
+* #### PHP - 7.4
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]: Why 7.4?

The plugin still needs to run against PHP 5.6.
```suggestion
* PHP - 5.6–8.0
```

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+## Setting up your environment
+
+### Minimum required versions
+
+* #### Node.js - LTS
+	Node.js is used in the build process of the Parse.ly plugin. If it's not already installed on your system, you can [visit the Node.js website and install the latest Long Term Support (LTS) version.](https://nodejs.org/)
+* #### PHP - 7.4 
+	  There are multiple ways to install PHP on your operating system. You can check out the [official installation instructions from the PHP project's website.](https://www.php.net/manual/en/install.php)
+
+* #### Composer - 2.0
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
* Composer - 1.x (but 2.x recommended).
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+## Reporting Bugs
+
+Please search the [repo's issues](https://github.com/Parsely/wp-parsely/issues) to see if your issue has been reported already and if so, comment on that issue instead of opening a new one.
+
+When creating a new issue, please add specific steps to reproduce the problem, upload any relevant screenshots, and describe what happened and what you expected would happen instead.
+
+## Setting up your environment
+
+### Minimum required versions
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md:13]: Missing what version of WP the plugin is tested against.

Technically, we only check against the latest version, but we claim to support WP 4.0 (we should add that to the CI matrix).

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+* #### Node.js - LTS
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]: Currently, [the plugin requires 14, and NPM 7](https://github.com/Parsely/wp-parsely/blob/514fc01bdcde57987c1166f4c622fd53ced17f2e/package.json#L7-L8).

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	Node.js is used in the build process of the Parse.ly plugin. If it's not already installed on your system, you can [visit the Node.js website and install the latest Long Term Support (LTS) version.](https://nodejs.org/)
+* #### PHP - 7.4 
+	  There are multiple ways to install PHP on your operating system. You can check out the [official installation instructions from the PHP project's website.](https://www.php.net/manual/en/install.php)
+
+* #### Composer - 2.0
+	The Parse.ly plugin includes several packages that require Composer, the PHP package manager.  You can view the [composer.json](https://github.com/Parsely/wp-parsely/blob/develop/composer.json) file for a full list of packages.  You can install Composer through Homebrew on mac os `brew install composer`.  If you don't have access to Homebrew you can view instructions for how to install composer on the [composer website](https://getcomposer.org/download/).
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
	The Parse.ly plugin includes several packages that require Composer, the PHP package manager. You can view the [composer.json](https://github.com/Parsely/wp-parsely/blob/develop/composer.json) file for a full list of packages. You can install Composer through Homebrew on macOS: `brew install composer`. If you don't have access to Homebrew you can view instructions for how to install Composer on the [Composer website](https://getcomposer.org/download/).
```

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+### Installing Dependencies
+Once you have Node.js, PHP, and Composer installed locally, you will need to run `composer install` in the main plugin directory to install the plugins dependencies. This will install the packages needed to run things like Unit Tests and check Coding Standards.
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
Once you have Node.js, PHP, and Composer installed locally, you will need to run `composer install` in the main plugin directory to install the dependencies of the plugin needed to run tests and check coding standards.
```

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
* Node.js - LTS
```

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+# Contributing patches and new features
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
## Contributing Patches and New Features
```

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+## Branches
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
### Branches
```

File: .github/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+Ongoing development will be done in the `develop` branch with merges done into `trunk` once considered stable.
+
+To contribute an improvement to this project, fork the repo and open a pull request to the `develop` branch. Alternatively, if you have push access to this repo, create a feature branch and then open an intra-repo PR from that branch to `develop`.
+
+# Code Standards
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
```

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+composer install
+```
+To lint your PHP code:
+```
+composer lint
+```
+To test your code with our Code Standards:
+```
+compose cs
+```
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
~~~suggestion
### Coding Standards
The Parse.ly plugin uses the PHP_CodeSniffer tool that is installed through Composer.  This plugin uses a [custom ruleset.](https://github.com/Parsely/wp-parsely/blob/develop/.phpcs.xml.dist)

For JavaScript we recommend installing ESLint. This plugin includes a [.eslintrc](https://github.com/Parsely/wp-parsely/blob/develop/.eslintrc) file that defines our coding standards. 

### Linting
If you haven't installed Composer, you will need to do that first.

To lint your PHP code:
```
composer lint
```

To check your code with our code standards:
```
compose cs
```
~~~

File: .github/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+# Unit Testing
+
+## PHPUnit
+
+### Pre-requisites
+- Install Composer
+- Install Composer dependencies with `composer install`
+- PHPUnit 4.x, 5.x, 6.x or 7.x
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
### Testing
```

No need to repeat about Composer for each section. 
The PHPUnit dependencies are handled by composer.json, so no need to specify the versions here.

File: .github/CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+### Running the tests
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
```

File: .github/CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+
+First, you'll need to install the tests by running the install script:
+1. Navigate to the main plugin directory.
+1. You'll need to have a local database setup and have the database name, username, password, and host ready.  
+1. Run the test install script using the database information.
+	```
+	./bin/install-wp-tests.sh <db-name> <db-user> <db-pass> [db-host] [wp-version] [skip-database-creation]"
+	```
+
+Then, you can use composer to run the tests from your terminal.
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
Then, you can use Composer to run the tests from your terminal.
```

File: .github/CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+To run the single-site tests: 
+```
+composer test
+```
+
+To run the multisite tests:
+```
+composer test-ms
+```
+
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md:103]:
~~~suggestion

To run with code coverage:
```
composer coverage
```
~~~

File: .github/CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+# Building Included Assets
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]:
```suggestion
## Building Included Assets
```

File: .github/CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+
+JavaScript files that are included in the released plugin are built with the
+[wp-scripts tool](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/).
+
+Here's how to get started making changes to them:
+
+1. Install the LTS release of node (if you don't have it already)
+1. Update to the latest version of npm
```
Comment (GaryJones) [inline .github/CONTRIBUTING.md]: These are already covered above.

```suggestion
```

**Review by GaryJones — DISMISSED** [review]
Many of the suggested changes are to keep the heading levels correct - just like HTML, there should only be a single H1 (`#`) in the document at the top, where as there were multiple before. The rest of the headings should come sequentially per the document structure. No need for headings on the list items.

Not included, but maybe consider splitting up what is the plugin's runtime compatibilities (PHP 5.6–8.0, WP 4.0-latest), from the environment requirements.

**Review by jblz — APPROVED** [review]

**Pull request merged**